tools/generate_hotkeys.py

```diff
@@ -1,39 +1,141 @@
 #!/usr/bin/env python3
+import argparse
+import re
+from collections import defaultdict
 from pathlib import Path, PurePath
+from typing import Dict, List, Tuple
 
 from zulipterminal.config.keys import HELP_CATEGORIES, KEY_BINDINGS
 
 
-categories = {
-    category: {
-        item["help_text"]: item["keys"]
-        for item in KEY_BINDINGS.values()
-        if item["key_category"] == category
-    }
-    for category in HELP_CATEGORIES.keys()
-}
+# Path to original docs/hotkeys.md
 OUTPUT_FILE = Path(__file__).resolve().parent.parent / "docs" / "hotkeys.md"
+
+# Name of the script
 SCRIPT_NAME = PurePath(__file__).name
 
-with open(OUTPUT_FILE, "w") as mdFile:
-    mdFile.write(
-        f"<!--- Generated automatically by tools/{SCRIPT_NAME} -->\n"
-        "<!--- Do not modify -->\n\n# Hot Keys\n"
-    )
-    for action in categories.keys():
-        mdFile.write(
-            f"## {HELP_CATEGORIES[action]}\n"
-            "|Command|Key Combination|\n"
-            "| :--- | :---: |\n"
-        )
-        for help_text, key_combinations_list in categories[action].items():
-            various_key_combinations = " / ".join(
-                [
-                    " + ".join([f"<kbd>{key}</kbd>" for key in key_combination.split()])
-                    for key_combination in key_combinations_list
-                ]
+# Exclude keys from duplicate keys checking
+KEYS_TO_EXCLUDE = ["q", "e", "m", "r"]
+
+
+def main(check_only: bool = False) -> None:
+    if check_only:
+        lint_hotkeys_file()
+    else:
+        generate_hotkeys_file()
+
+
+def lint_hotkeys_file() -> None:
+    """
+    First compare existing hotkeys file, if in sync then lint for key description and key duplication.
+    """
+    hotkeys_file_string = get_hotkeys_file_string()
+    if compare_hotkeys_file(hotkeys_file_string):
+        # To lint keys description and duplicate keys
+        error_flag = 0
+        help_text_style = re.compile(r"^[a-zA-Z\s/()',&@#:_-]*$")
+        categories = read_help_categories()
+        for action in HELP_CATEGORIES.keys():
+            check_duplicate_keys_list: List[str] = []
+            for help_text, key_combinations_list in categories[action]:
+                check_duplicate_keys_list.extend(key_combinations_list)
+                various_key_combinations = various_key_combination(
+                    key_combinations_list
+                )
+                # Check description style
+                if not re.match(help_text_style, help_text):
+                    print(
+                        f"Description - ({help_text}) for key combination - [{various_key_combinations}] should contain only alphabets, spaces and special characters except ."
```

Comment:
Here, the `various_key_combinations` is printed with the `kbd` tags. I think it would be more readable if it is written without the tags.